### ISSUE TITLE
Icon size changes when pulling node from palette into view

### ISSUE THREAD

**Issue by ralphwetzel** [node-red/node-red#4490]
### Current Behavior

![icon_size](https://github.com/node-red/node-red/assets/16342003/4f026394-1301-4e5a-baf2-a9424e1f7467)

I was just creating a contrib node.
As icon of this node I'm using an (oversized) `.png` graphics.
The size of the nodes icon displayed increases when the node is pulled from the palette and dropped into the view.

### Environment

- Node-RED version: v3.1.1
- Node.js version: v16.18.1
- Platform/OS: macOS Sonoma 14.2
- Browser: Chrome

**Comment by knolleary** [comment]
A consequence of the different rendering code used in the sidebar (all html/css) and the main workspace (svg).

**Comment by ralphwetzel** [comment]
Might be - yet doesn't need to.
Looks like there's a different `width` setting active for the two cases.
I'll check if there's a potential fix w/ reasonable effort...

**Comment by ralphwetzel** [comment]
**Analysis:**

Icons in the palette get the `red-ui-palette-icon` class.
https://github.com/node-red/node-red/blob/0e8d3127944a91f39fb49bcb4b388b184ce11edd/packages/node_modules/%40node-red/editor-client/src/js/ui/utils.js#L1311-L1312
For the palette, `red-ui-palette-icon` is defined with `width: 20px`.
https://github.com/node-red/node-red/blob/0e8d3127944a91f39fb49bcb4b388b184ce11edd/packages/node_modules/%40node-red/editor-client/src/sass/palette.scss#L184-L191
Icon files, in case they don't carry the ending`.svg`, are scaled to "largestEdge max 30 pixels":
https://github.com/node-red/node-red/blob/0e8d3127944a91f39fb49bcb4b388b184ce11edd/packages/node_modules/%40node-red/editor-client/src/js/ui/view.js#L4150-L4162

**Proposal:**

To keep a consistent appearance, this should rather be "largestEdge max 30 pixels && width max 20 pixels"

```
img.onload = function() {
    if (!iconUrl.match(/\.svg$/)) {
        var largestEdge = Math.max(img.width,img.height);
        var scaleFactor = 1;
        if (largestEdge > 30) {
            scaleFactor = 30/largestEdge;
        }
        var width = img.width * scaleFactor;
        if (width > 20) {
            scaleFactor *= 20/width;
            width = 20;
        }
        var height = img.height * scaleFactor;
        icon.attr("width",width);
        icon.attr("height",height);
        icon.attr("x",15-width/2);
        icon.attr("y", (30-height)/2);  // to center img vertically
    }
```

![icon_size_2](https://github.com/node-red/node-red/assets/16342003/8677ffa5-2ee8-4505-a27a-b4f922e0b8a4)

PR?

**Comment by dceejay** [comment]
(though generally we suggest white only icons... ;-)

**Comment by knolleary** [comment]
Fixed by #4491
